src/ducks/transactions/queries.js

```diff
@@ -39,32 +42,32 @@ export const makeFilteredTransactionsConn = options => {
           const group = groups.data.find(g => g._id === filteringDoc._id)
           accounts = group ? group.accounts.raw : []
         }
-        indexFields = ['date', 'account']
+        indexFields = [dateAttribute, 'account']
```

Comment:
I don't find anything related to `applicationDate` in our doctype, can you add it? https://github.com/cozy/cozy-doctypes/blob/master/docs/io.cozy.bank.md

Reply:
Also we currently added warmupqueries in banks https://github.com/cozy/cozy-banks/blob/3d02f1b7e5c152f606ac5034e725c9c564e18a0f/src/ducks/client/links.js#L35

Should we do the same with this `applicationDate` index? When is it used? By default?

Reply:
pour le doctype : https://github.com/cozy/cozy-doctypes/pull/178

Reply:
Attention à l'ajout des warmup queries. Ça peut prendre pas mal de temps à créer et ça peut occuper pas mal de temps le CPU du périphérique. C'est pas une solution magique. Et surtout, aujourd'hui tant que toutes les requêtes de préchauffe ne sont pas faites, l'application va utiliser la stack plutôt que la base locale. Ce qui peut potentiellement gâcher un peu la première expérience offline. (Une solution un peu plus "magique" c'est de changer l'adapteur PouchDB comme ce qu'on vient de faire sur Drive https://github.com/cozy/cozy-drive/pull/2399 regarder les temps d'indexation). @paultranvan le fera certainement sur Banks il faut qu'on en discute. 

Donc ma question est la suivante : Dans quel cas on utilise l'index sur `applicationDate` dans l'application ? Est-ce l'index par défaut à utiliser ? Est-ce qu'il est accessible seulement après quelques manipulations de l'utilisateur ?

Reply:
On l'utilise par défaut sur la page de `analyse > catégorisation`, mais qui n'est pas la page par défaut de banks. Il faut donc que le user clic sur `Analyse`.

Si on a des contraintes CPU et autre, je pense que ce n'est donc pas nécessaire de les mettre en warmup. Il n'y a pas 36 pages dans banks donc on a une forte probabilité que le user aille sur cette page d'analyse, mais ça ne devrait pas être au détriment de la première utilisation. Sur la page analyse, on a les spinner et tous les mécanismes pour gérer correctement les requêtes...

Qu'en penses-tu, je supprime donc ?

Reply:
Je teste pour voir les performances, si jamais on a des surcharges CPU je ferai une autre PR pour l'enlever.

Reply:
> on a une forte probabilité que le user aille sur cette page d'analyse, mais ça ne devrait pas être au détriment de la première utilisation.

Si on ne paye pas le coût d'indexation à la première utilisation de Banks, on le payera à la première utilisation de catégorisation.
Dans tous les cas, il faut payer le temps d'indexation à un moment, i.e. Couch/Pouch qui map tous les docs pour construire un B-Tree sur la base des attributs déclarés. Le warmup permet de faire ça à l'init et d'utiliser la stack en attendant. Sinon, ça veut dire que l'index sera calculé au moment où l'utilisateur ira dans catégorisation la 1ère fois, ce qui peut être assez lent s'il y a beaucoup de docs / un adapteur Pouch peu performant (avec indexeddb ça s'améliore).

Sur un autre sujet, quand on indexe plusieurs attributs, il vaut mieux indexer en 1er l'attribut sur lequel on aura un prédicat d'égalité, et en dernier celui avec un prédicat d'intervalle, typiquement une date, cf https://docs.cozy.io/en/tutorials/data/advanced/#indexes-performances-and-design. Du coup, il vaudrait mieux avoir `indexFields = [account, applicationDate]` dans cet ordre

Reply:
Oui tout à fait il faut toute façon payer le coup à un moment donnée. Mais en terme d'UX l'approche qui a fait ses preuves il me semble est plutôt celle ou on a un premier chargement d'app rapide, puis des loader dans l'app, et non un gros chargement initial.
En fait l'idéal serait peut-être d'avoir un compromis : on warmup les requêtes de la première page de l'app, et sur cette page on warmup les requêtes des autres pages de l'app.

Ok merci pour la précision concernant l'ordre des index 👍

Reply:
@paultranvan On a laissé les warmup pour info, mais on va tester pour voir ce que ça donne car aujourd'hui les warmup se font toutes en // (https://github.com/cozy/cozy-client/blob/e82646c0a73a0f2873010ee2da4c13e651b9c5af/packages/cozy-pouch-link/src/PouchManager.js#L295-L302) et j'ai peur qu'en ayant 6 ou 8 requêtes en // on flingue le device. 

On devrait peut-être avoir un warmup par requête et non pas par doctype comme actuellement et potentiellement faire de manière séquentielle ?